runelite-client/src/main/java/net/runelite/client/plugins/loginscreen/LoginScreenPlugin.java

```diff
@@ -287,6 +288,11 @@ private void overrideLoginScreen()
 				}
 			}
 		}
+		else if (config.loginScreen() == LoginScreenOverride.RANDOM)
+		{
+			LoginScreenOverride rand = LoginScreenOverride.values()[new Random().nextInt(LoginScreenOverride.values().length - 1)];
```

Comment:
`length - 1` is not a clear indicator of the logic happening on this line. Furthermore, it would give greater weight to one login screen given that `OFF` is often/always synonymous with some other login screen, usually `NORMAL`.

Instead, you can filter the available login screens to a list of only the ones with non-null file names, then select a random entry from there.

Reply:
How does that look?